### ISSUE TITLE
length(rows) != length(cols) in add_linear_constraints!

### ISSUE THREAD

**Issue by IssamT** [JuliaOpt/LinQuadOptInterface.jl#12]
This call

https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/master/src/constraints/scalaraffine.jl#L37

seems inconsistant with the interface contract

https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/master/src/solver_interface.jl#L153

**Comment by odow** [comment]
Looks like I made a mistake in the docs.

It should actually be the starting index of each row.

See implementation in Gurobi:
https://github.com/JuliaOpt/Gurobi.jl/blob/b8a53ea4c5073220d245b0eedd5572ed33200296/src/MOIWrapper.jl#L129
https://github.com/JuliaOpt/Gurobi.jl/blob/b8a53ea4c5073220d245b0eedd5572ed33200296/src/grb_constrs.jl#L35

**Comment by odow** [comment]
The docs should be an expanded version of this
```julia
"""
    add_linear_constraints!(m, row_begins::Vector{Int}, cols::Vector{Int},
        coefs::Vector{Float64},
        sense::Vector{Cchar}, rhs::Vector{Float64})::Void

Adds linear constraints of the form `Ax (sense) rhs` to the model `m`.

The A matrix is given in compressed sparse row format. 
... TODO -- flesh this out ...

### Example

To add the constraints:
    1.0x + 2.0y <= 3.0
            4.0y >= 5.0,
the call is:
    add_linear_constraints(m, 
        [1, 3], 
        [1, 2, 2], 
        [1.0, 2.0, 4.0], 
        ["<=", ">="], 
        [3.0, 5.0]
    )
"""
```

**Comment by IssamT** [comment]
Thanks for the explanation @odow

**Comment by odow** [comment]
Closed by #17
